drivers/gpio/gpio_sam0_eic.c

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright (c) 2018 Google LLC.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gpio.h>
+#include <soc.h>
+
+#if CONFIG_EIC_SAM0_BASE_ADDRESS
```

Comment:
when would this not be defined?

Reply:
I'm open to suggestions here.  If you had a minimal system and didn't want GPIO interrupts then you could drop the EIC from Device Tree and save ~1 KiB.

Reply:
Not sure I understand, what the issue with GPIO interrupts.  Guessing this has something to do with use of EIC vs normal GPIO or something.

Reply:
Ah, I see.  The SAM0 has a PORT peripheral that sets the pin direction, level, pull up etc and a separate shared EIC peripheral that does edge detection and generates interrupts.  The PORT pins are muxed onto the EIC inputs, so for example PA0, PA16, PB0, and PB16 all map to EIC channel 0 and hence only one can be used at once.

Reply:
After thinking about it, there's no real need to support configuring out the EIC so I'll drop this #if